### PR TITLE
refactoring internal APIs and domain model

### PR DESCRIPTION
**Pull request by battermann**

### CODE REVIEW

File: src/MidiConversions.elm [hunk 1]
```diff
@@ -1,43 +1,21 @@
-module MidiConversions exposing (toBase64EncodedMidi, toDataString, toMidiNumber)
+module MidiConversions exposing (toMidiNumber, createDataLink)
 
-import Types exposing (..)
+import Pitch exposing (..)
 import Array exposing (Array, toList)
-import Midi.Types as Midi
```
Comment (jarrodu) [inline src/MidiConversions.elm]: :+1:

File: src/MidiConversions.elm [hunk 2]
```diff
@@ -74,30 +52,27 @@ accidentalToInt accidental =
...
-    , midiData = (encode (toMidi bytes))
-    }
+concatParts : String -> String -> String -> String
+concatParts contentType contentTransferEncoding midiData =
+    "data:" ++ contentType ++ ";" ++ contentTransferEncoding ++ "," ++ midiData
+
+
+createDataLink : Array Pitch -> String
+createDataLink =
+    toMidi >> BinaryBase64.encode >> concatParts "audio/midi" "base64"
```
Comment (jarrodu) [inline src/MidiConversions.elm]: I like this style a bit better than using '|>'. Is this called "point free"? It looks like function composition.

File: src/MidiConversions.elm [hunk 3]
```diff
@@ -74,30 +52,27 @@ accidentalToInt accidental =
...
+                , ( 2, Midi.Types.NoteOff 0 midiNumber 0 )
                 ]
             )
         |> flatten
-        |> Midi.SingleTrack 4
+        |> Midi.Types.SingleTrack 4
         |> recording
 
 
-toBase64EncodedMidi : Array Pitch -> MidiEncoding
```
Comment (jarrodu) [inline src/MidiConversions.elm]: Thanks. I thought this was a bit overkill too.

**Comment by battermann** [comment]
Not sure about `require` vs `import`. Just wanted to make it more consistent.

File: src/MidiConversions.elm (hunk 2 above)
Comment (battermann) [reply src/MidiConversions.elm]: Exactly, it's reversed function composition.

**Pull request merged**